ecs-cli/modules/cli/logs/logs_app_test.go

```diff
@@ -389,3 +392,80 @@ func TestLogsRequestTaskNotFound(t *testing.T) {
 	assert.Error(t, err, "Expected error getting logs")
 }
+
+/* Create Logs */
+
+func TestCreateLogGroups(t *testing.T) {
+	taskDef := dummyTaskDef([]*ecs.ContainerDefinition{
+		dummyContainerDef(logRegion1, logGroup1, logPrefix1, "awslogs", containerName, containerImage),
+	})
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockLogFactory := mock_cloudwatchlogs.NewMockLogClientFactory(ctrl)
+	mockLogClient := mock_cloudwatchlogs.NewMockClient(ctrl)
+
+	gomock.InOrder(
+		mockLogFactory.EXPECT().Get(logRegion1).Return(mockLogClient),
+		mockLogClient.EXPECT().CreateLogGroup(gomock.Any()),
+	)
+
+	err := CreateLogGroups(taskDef, mockLogFactory)
+	assert.NoError(t, err, "Unexpected error in call to CreateLogGroups()")
+}
+
+func TestCreateLogGroupsTwoContainers(t *testing.T) {
+	taskDef := dummyTaskDef([]*ecs.ContainerDefinition{
+		dummyContainerDef(logRegion1, logGroup1, logPrefix1, "awslogs", containerName, containerImage),
+		dummyContainerDef(logRegion2, logGroup2, logPrefix1, "awslogs", containerName, containerImage),
+	})
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockLogFactory := mock_cloudwatchlogs.NewMockLogClientFactory(ctrl)
+	mockLogClient := mock_cloudwatchlogs.NewMockClient(ctrl)
+
+	gomock.InOrder(
+		mockLogFactory.EXPECT().Get(logRegion1).Return(mockLogClient),
+		mockLogClient.EXPECT().CreateLogGroup(gomock.Any()),
+		mockLogFactory.EXPECT().Get(logRegion2).Return(mockLogClient),
+		mockLogClient.EXPECT().CreateLogGroup(gomock.Any()),
+	)
+
+	err := CreateLogGroups(taskDef, mockLogFactory)
+	assert.NoError(t, err, "Unexpected error in call to CreateLogGroups()")
+}
+
+func TestCreateLogGroupsWrongDriver(t *testing.T) {
+	taskDef := dummyTaskDef([]*ecs.ContainerDefinition{
+		dummyContainerDef(logRegion1, logGroup1, logPrefix1, "catsanddogslogger", containerName, containerImage),
+	})
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockLogFactory := mock_cloudwatchlogs.NewMockLogClientFactory(ctrl)
+
+	err := CreateLogGroups(taskDef, mockLogFactory)
+	assert.Error(t, err, "Expected error in call to CreateLogGroups()")
+}
+
+func TestCreateLogGroupsLogGroupAlreadyExists(t *testing.T) {
+	taskDef := dummyTaskDef([]*ecs.ContainerDefinition{
+		dummyContainerDef(logRegion1, logGroup1, logPrefix1, "awslogs", containerName, containerImage),
+	})
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockLogFactory := mock_cloudwatchlogs.NewMockLogClientFactory(ctrl)
+	mockLogClient := mock_cloudwatchlogs.NewMockClient(ctrl)
+
+	alreadyExistsErr := awserr.New(cloudwatchlogs.ErrCodeResourceAlreadyExistsException, "Resource Already Exists Exception", nil)
+
+	gomock.InOrder(
+		mockLogFactory.EXPECT().Get(logRegion1).Return(mockLogClient),
+		mockLogClient.EXPECT().CreateLogGroup(gomock.Any()).Return(alreadyExistsErr),
+	)
+
+	err := CreateLogGroups(taskDef, mockLogFactory)
+	assert.NoError(t, err, "Unexpected error in call to CreateLogGroups()")
+}
```

Comment:
Can you add a test in which the client throws a non-"ErrCodeResourceAlreadyExistsException" error when you call "CreateLogGroup"?

Reply:
Sure!